### PR TITLE
feat: add mock resource page by parsing index.md created by Netlify-CMS

### PR DESCRIPTION
**Pull request by ljc1991**
# Description

- Finish page file parser.
- Add mock resource page.

## Major

- Use page parser to parse `index.md`
- WIP: add `remark` component to replace `dangerouslySetInnerHTML`
- Questions: How to create `Navbar` and `<Head>`, `<Script>` from Netlify-CMS

# Impaction

## Screenshots

| Scenarios  | Before | After |
| ---------- | ------ | ----- |
| Compare to original homepage | 截圖 2023-09-06 16 08 12 截圖 2023-09-06 16 08 20 截圖 2023-09-06 16 08 31|截圖 2023-09-06 15 45 35 截圖 2023-09-06 15 45 46 截圖 2023-09-06 15 45 54|

## Performance
N/A

### CODE REVIEW

**Comment by ljc1991** [comment]
deploy failed orz...

due to lack of `en/resource` page?

**Comment by ben196888** [comment]
> deploy failed orz...
> 
> due to lack of `en/resource` page?

I think so. The logs shows the resource page export failure.

```
3:32:28 PM: Error: Export encountered errors on following paths:
3:32:28 PM: 	/resource: /en/resource
3:32:28 PM:     at /opt/build/repo/homepage/node_modules/next/dist/export/index.js:425:19
3:32:28 PM:     at processTicksAndRejections (node:internal/process/task_queues:96:5)
3:32:28 PM:     at async Span.traceAsyncFn (/opt/build/repo/homepage/node_modules/next/dist/trace/trace.js:79:20)
3:32:28 PM:     at async /opt/build/repo/homepage/node_modules/next/dist/build/index.js:1403:21
3:32:28 PM:     at async Span.traceAsyncFn (/opt/build/repo/homepage/node_modules/next/dist/trace/trace.js:79:20)
3:32:28 PM:     at async /opt/build/repo/homepage/node_modules/next/dist/build/index.js:1261:17
3:32:28 PM:     at async Span.traceAsyncFn (/opt/build/repo/homepage/node_modules/next/dist/trace/trace.js:79:20)
3:32:28 PM:     at async Object.build [as default] (/opt/build/repo/homepage/node_modules/next/dist/build/index.js:72:29)
```

File: homepage/src/pages/resource.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    props: {
+      page,
+    },
+  };
+};
+
+export default function Resource({ page }) {
+  return (
+    <>
+      {page.data['layout_list'].map((layout) => {
```
Comment (ben196888) [inline homepage/src/pages/resource.jsx:29]: `page.data.layout_list` would be undefined for some case. We shall change to `page.data['layout_list']?.map`.

File: homepage/src/pages/resource.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                title={layout?.title}
+                subtitle={layout?.subtitle}
+                heroImage={layout?.heroImage}
+                highlights={layout?.highlights}
+              />
+            );
+          case 'layout_section':
+            if (layout.columns?.length === 2) {
+              const columns = layout.columns?.map((column) => {
+                return [column.title?.toString(), column.text?.toString()];
```
Comment (ben196888) [inline homepage/src/pages/resource.jsx:44]: Do we need `.toString()` for column.title and column.text? Aren't these values string type?

File: homepage/src/pages/resource.jsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import Headline from '../components/headline';
+
+import { fetchPage } from '../lib/fetchPage';
+
+/**
+ *
+ * @type {import('next').GetStaticProps}
+ */
+export const getStaticProps = async ({ locale }) => {
+  const page = fetchPage(locale, 'index');
```
Comment (ben196888) [inline homepage/src/pages/resource.jsx:17]: I've created the basic resource page. We can replace the `index` w/ `resource` safely.

Comment (ljc1991) [reply homepage/src/pages/resource.jsx:17]: ok!

File: homepage/src/pages/resource.jsx (hunk 1 above)
Comment (ljc1991) [reply homepage/src/pages/resource.jsx:29]: fixed.

Comment (ljc1991) [reply homepage/src/pages/resource.jsx:29]: fixed.

File: homepage/src/pages/resource.jsx (hunk 2 above)
Comment (ljc1991) [reply homepage/src/pages/resource.jsx:44]: lol

File: homepage/src/pages/resource.jsx (hunk 3 above)
Comment (ljc1991) [reply homepage/src/pages/resource.jsx:17]: ok!

**Comment by ben196888** [comment]
🍐 w/ @ljc1991 to fix the paragraph styling and markdown workaround issue.

**Pull request merged**